corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -314,7 +314,7 @@ function DateTimeEntryBase(question, options) {
                 }
                 self.answer(moment(newDate).format(self.serverFormat));
             }
-        }).val(moment(self.answer()).format(self.clientFormat));
+        }).val(self.answer() ? moment(self.answer()).format(self.clientFormat) : self.answer());
```

Comment:
essentially this says if it's null or undefined, leave it, if not, try and convert it to a date